checks.d/mysql.py

```diff
@@ -32,78 +33,341 @@
+    'Handler_read_prev': ('mysql.performance.handler_read_prev', RATE),
+    'Handler_read_rnd': ('mysql.performance.handler_read_rnd', RATE),
+    'Handler_read_rnd_next': ('mysql.performance.handler_read_rnd_next', RATE),
+    'Handler_rollback': ('mysql.performance.handler_rollback', RATE),
+    'Handler_update': ('mysql.performance.handler_update', RATE),
+    'Handler_write': ('mysql.performance.handler_write', RATE),
+    'Opened_tables': ('mysql.performance.opened_tables', RATE),
+    'Qcache_total_blocks': ('mysql.performance.qcache_total_blocks', GAUGE),
+    'Qcache_free_blocks': ('mysql.performance.qcache_free_blocks', GAUGE),
+    'Qcache_free_memory': ('mysql.performance.qcache_free_memory', GAUGE),
+    'Qcache_not_cached': ('mysql.performance.qcache_not_cached', RATE),
+    'Qcache_queries_in_cache': ('mysql.performance.qcache_queries_in_cache', GAUGE),
+    'Select_full_join': ('mysql.performance.select_full_join', RATE),
+    'Select_full_range_join': ('mysql.performance.select_full_range_join', RATE),
+    'Select_range': ('mysql.performance.select_range', RATE),
+    'Select_range_check': ('mysql.performance.select_range_check', RATE),
+    'Select_scan': ('mysql.performance.select_scan', RATE),
+    'Sort_merge_passes': ('mysql.performance.sort_merge_passes', RATE),
+    'Sort_range': ('mysql.performance.sort_range', RATE),
+    'Sort_rows': ('mysql.performance.sort_rows', RATE),
+    'Sort_scan': ('mysql.performance.sort_scan', RATE),
+    'Table_locks_immediate': ('mysql.performance.table_locks_immediate', RATE),
+    'Threads_cached': ('mysql.performance.threads_cached', GAUGE),
+    'Threads_created': ('mysql.performance.threads_created', GAUGE)
+}
+
+# Status Vars added in Mysql 5.6.6
+OPTIONAL_STATUS_VARS_5_6_6 = {
+    'Table_open_cache_hits': ('mysql.performance.table_cache_hits', RATE),
+    'Table_open_cache_misses': ('mysql.performance.table_cache_misses', RATE),
+}
+
+# Will collect if [FLAG NAME] is True
+OPTIONAL_INNODB_VARS = {
+    'Innodb_active_transactions': ('mysql.innodb.active_transactions', GAUGE),
+    'Innodb_buffer_pool_bytes_data': ('mysql.innodb.buffer_pool_data', GAUGE),
+    'Innodb_buffer_pool_pages_data': ('mysql.innodb.buffer_pool_pages_data', GAUGE),
+    'Innodb_buffer_pool_pages_dirty': ('mysql.innodb.buffer_pool_pages_dirty', GAUGE),
+    'Innodb_buffer_pool_pages_flushed': ('mysql.innodb.buffer_pool_pages_flushed', RATE),
+    'Innodb_buffer_pool_pages_free': ('mysql.innodb.buffer_pool_pages_free', GAUGE),
+    'Innodb_buffer_pool_pages_total': ('mysql.innodb.buffer_pool_pages_total', GAUGE),
+    'Innodb_buffer_pool_read_ahead': ('mysql.innodb.buffer_pool_read_ahead', RATE),
+    'Innodb_buffer_pool_read_ahead_evicted': ('mysql.innodb.buffer_pool_read_ahead_evicted', GAUGE),
+    'Innodb_buffer_pool_read_ahead_rnd': ('mysql.innodb.buffer_pool_read_ahead_rnd', GAUGE),
+    'Innodb_buffer_pool_wait_free': ('mysql.innodb.buffer_pool_wait_free', GAUGE),
```

Comment:
This is a counter (http://dev.mysql.com/doc/refman/5.7/en/server-status-variables.html#statvar_Innodb_buffer_pool_wait_free) -> should probably be a `RATE` too, unless we want the value since the server started.